### PR TITLE
Xray joins 14793 mk3

### PR DESCRIPTION
**Pull request by howonlee**
Pursuant to #14793. This splices in the joins automagically post-hoc. I attempted a way to change the root so it stored join data and put them in that way but it turned into a boiling-the-ocean task. This method seems to work. No separation into parts, just one PR that makes the thing work

Also there was a bug in the cypress test itself, lol

Also I was curious as to whether it would mutate any of the other example runs of the xrays and it seems no.

### CODE REVIEW

**Comment by howonlee** [comment]
Cardinality of the cards generated from fields is changing depending upon inputs, driver, the weather, how i'm feeling, the exchange rate of rupees in bangalore... I think I will just dictate a range

File: src/metabase/automagic_dashboards/core.clj [hunk 1]
```diff
@@ -1163,13 +1164,36 @@
                    (:full-name root))
                  (tru "where {0}" (humanize-filter-value root cell-query))]))
 
+(defn- key-in?
+  "Recursively finds key in coll, returns true or false"
+  [coll k]
+  (boolean (let [coll-zip (zip/zipper coll? #(if (map? %) (vals %) %) nil coll)]
+    (loop [x coll-zip]
+      (when-not (zip/end? x)
+        (if-let [v (k (zip/node x))] true (recur (zip/next x))))))))
```
Comment (dpsutton) [inline src/metabase/automagic_dashboards/core.clj:1173]: Can joins be hidden in lower levels? Or are they always at `[:query :joins]`?

Comment (howonlee) [reply src/metabase/automagic_dashboards/core.clj:1173]: yes

https://github.com/metabase/metabase/blob/be98b105ac914bba6274049beff691ae9058defa/shared/src/metabase/mbql/schema.cljc#L1032

Comment (dpsutton) [reply src/metabase/automagic_dashboards/core.clj:1173]: Also, probably more standard to use the match utilies on mbql
```clojure
match=> (match-one {:database 234,
                    :type :query,
                    :query
                    {:fields [[:field 33845 nil] [:field 33847 {:join-alias "f"}]],
                     :joins
                     [{:source-table 10804,
                       :condition [:= [:field 33844 nil] [:field 33848 {:join-alias "f"}]],
                       :strategy :left-join,
                       :alias "f"}],
                     :order-by [[:asc [:field 33845 nil]]],
                     :source-table 10803}}
                   {:joins [joins]} joins)
{:source-table 10804, :condition [:= [:field 33844 nil] [:field 33848 {:join-alias "f"}]], :strategy :left-join, :alias "f"}
```

Alth

Comment (dpsutton) [reply src/metabase/automagic_dashboards/core.clj:1173]: oh i see my confusion. Why is `splice-in` conditioned on the presence of `join-alias`? If the root has a join, wouldn't we want all of the cards to include that join?

Comment (howonlee) [reply src/metabase/automagic_dashboards/core.clj:1173]: there's other functionality to put in joins after the fact in pretty disparate portions of the xray codebase and I didn't want funky action-at-a-distance shenanigans

**Review by dpsutton — APPROVED** [review]

**Pull request merged**